### PR TITLE
Fix local_mic & input width issues in mic.py

### PR DESCRIPTION
**Pull request by jfremy**
Those commits fix 2 issues in the current dev branch:
- local_mic.py was not working anymore since the switch to audioengine because of the introduction of the listen function and change in the type of the return value.
- mic.py assumes that sample width is 1 byte while it's configured to 2 bytes. I update all (at least I think) computations to use self._input_width. The RMS computations and the silence timeout were doing it (so existing silence timeout was more 1.5s - I switched to 2s because from experience it seems to work ok while 3s seems a little long)

### CODE REVIEW

**Comment by Holzhaus** [comment]
Thanks, I'll look into it as soon as I finished my exams.

**Comment by jfremy** [comment]
Regarding the fix I made for pymad, I checked pymad source code and it always outputs stereo signed 16bits. So i modified the wave parameters to that.
Still it does not fix the popping noise when playing a wave file using pyaudio on my raspberry pi 2 (or B+ for that matter).

I've written a simple python test program that plays a wav file using pyaudio and it does not work correctly either (so it's not jasper's use of pyaudio that causes the problem). A file that plays perfectly with aplay has popping noise when playing with pyaudio.

I've tried to update the portaudio + pyaudio packages to the Debian Jessie version:
- libportaudio2_19+svn20140130-1_armhf.deb
- libportaudiocpp0_19+svn20140130-1_armhf.deb
- portaudio19-dev_19+svn20140130-1_armhf.deb
- python-pyaudio_0.2.8-1+b1_armhf.deb
It did not seem to change anything

I'm currently trying to isolate at which level it's happening if it's in libportaudio or in the python bindings.

**Comment by jfremy** [comment]
Issue seems to be in python-pyaudio.
I have C++ code using portaudio that works fine to play a wav file while using a python wav player has popping sound.

**Comment by jfremy** [comment]
Seems that increasing frames_per_buffer to 8192 fixes it.
Will add this to the changes

**Comment by jfremy** [comment]
But it does not work out of the box with the audioengine code.
I need to investigate but output is not always played to the end and the switch between output and input does not seem to work correctly

**Comment by jfremy** [comment]
Small update on my investigations on the behavior of pyaudio when frames_per_buffer is higher than 1024.
My sample python script I've been using seems to show that to have the file play until the end, you need to write 0s to the stream until you reach a multiple of frames_per_buffer.
Otherwise file won't be played until the end.

I don't know if this is specific to the raspbian version of pyaudio but that's what I'm seeing here.

**Comment by jfremy** [comment]
So playing with pyaudio I've been able to get better results but it's more of a workaround than anything else.
I can confirm that increasing the frames_per_buffer for play improves the audio quality on rpi.
However, when you do that, it does not play the audio until the end until you insert some padding at the end (something between 1 and 2 full buffers).

aplay or even a sample C program calling libportaudio works fine so it's in the pyaudio implementation that something goes wrong.

File: client/local_mic.py [hunk 1]
```diff
@@ -18,7 +18,10 @@ def wait_for_keyword(self, keyword="JASPER"):
     def active_listen(self, timeout=3):
         input = raw_input("YOU: ")
         self.prev = input
-        return input
+        return [input]
+
+    def listen(self):
```
Comment (Holzhaus) [inline client/local_mic.py]: Codacy Issue found: [Missing method docstring](https://www.codacy.com/app/jasperproject/jasper-client/file/769710982/issues/source?bid=2081347&fileBranchId=2146790#l23)

File: client/mic.py [hunk 2]
```diff
@@ -128,9 +130,10 @@ def listen(self):
         self.wait_for_keyword(self._keyword)
         return self.active_listen()
 
-    def active_listen(self, timeout=3):
+    def active_listen(self, timeout=2):
```
Comment (Holzhaus) [inline client/mic.py]: Codacy Issue found: [Missing method docstring](https://www.codacy.com/app/jasperproject/jasper-client/file/769710984/issues/source?bid=2081347&fileBranchId=2146790#l133)

**Comment by Holzhaus** [comment]
> So playing with pyaudio I've been able to get better results but it's more of a workaround than anything else.

I added a new audio engine based on alsa. You could try that, maybe that fixes the issue.

just update and put `audio_engine: alsa` into your `profile.yml`.

**Pull request closed without merging**